src/main/java/org/deepsymmetry/beatlink/UpdateSocketConnection.java

```diff
@@ -0,0 +1,449 @@
+        }
+    }
+
+    /**
+     * Get the set of media details listeners that are currently registered.
+     *
+     * @return the currently registered details listeners
+     */
+    public Set<MediaDetailsListener> getMediaDetailsListeners() {
+        // Make a copy so callers get an immutable snapshot of the current state.
+        return Collections.unmodifiableSet(new HashSet<MediaDetailsListener>(detailsListeners));
+    }
+
+    /**
+     * Send a media details response to all registered listeners.
+     *
+     * @param details the response that has just arrived
+     */
+    public void deliverMediaDetailsUpdate(final MediaDetails details) {
+        for (MediaDetailsListener listener : getMediaDetailsListeners()) {
+            try {
+                listener.detailsAvailable(details);
+            } catch (Throwable t) {
+                logger.warn("Problem delivering media details response to listener", t);
+            }
+        }
+    }
+
+
+
+    /**
+     * Given an update packet sent to us, create the appropriate object to describe it.
+     *
+     * @param packet the packet received on our update port
+     * @return the corresponding {@link DeviceUpdate} subclass, or {@code nil} if the packet was not recognizable
+     */
+    private DeviceUpdate buildUpdate(DatagramPacket packet) {
+        final int length = packet.getLength();
+        final Util.PacketType kind = Util.validateHeader(packet, UPDATE_PORT);
+
+        if (kind == null) {
+            logger.warn("Ignoring unrecognized packet sent to update port.");
+            return null;
+        }
+
+        switch (kind) {
+            case MIXER_STATUS:
+                if (length != 56) {
+                    logger.warn("Processing a Mixer Status packet with unexpected length " + length + ", expected 56 bytes.");
+                }
+                if (length >= 56) {
+                    return new MixerStatus(packet);
+                } else {
+                    logger.warn("Ignoring too-short Mixer Status packet.");
+                    return null;
+                }
+
+            case CDJ_STATUS:
+                if (length >= CdjStatus.MINIMUM_PACKET_SIZE) {
+                    return new CdjStatus(packet);
+
+                } else {
+                    logger.warn("Ignoring too-short CDJ Status packet with length " + length + " (we need " + CdjStatus.MINIMUM_PACKET_SIZE +
+                            " bytes).");
+                    return null;
+                }
+
+            case LOAD_TRACK_ACK:
+                logger.info("Received track load acknowledgment from player " + packet.getData()[0x21]);
+                return null;
+
+            case MEDIA_QUERY:
+                logger.warn("Received a media query packet, we don’t yet support responding to this.");
+                return null;
+
+            case MEDIA_RESPONSE:
+                if (packet.getLength() >= MediaDetails.MINIMUM_PACKET_SIZE) {
```

Comment:
@brunchboy this here was a bit confusing to me. The packets were `CdjStatus.MINIMUM_PACKET_SIZE` in length exactly, so I had to change this from `>` to `>=`. I have no idea why the packet size increased by one. Do you have any insight on this?